src/frequenz/client/base/channel.py

```diff
@@ -80,15 +80,7 @@ def parse_grpc_uri(
     port = parsed_uri.port or default_port
     match channel_type:
         case _grpchacks.GrpcioChannel:
-            target = f"{host}:{port}"
-            return (
-                _grpchacks.grpcio_secure_channel(
-                    target, _grpchacks.grpcio_ssl_channel_credentials()
-                )
-                if ssl
-                else _grpchacks.grpcio_insecure_channel(target)
-            )
+            return cast(ChannelT, _grpchacks.grpcio_create_channel(host, port, ssl))
         case _grpchacks.GrpclibChannel:
-            return _grpchacks.GrpclibChannel(host=host, port=port, ssl=ssl)
-        case _:
-            assert False, "Unexpected channel type: {channel_type}"
+            return cast(ChannelT, _grpchacks.grpclib_create_channel(host, port, ssl))
+    assert False, "Unexpected channel type: {channel_type}"
```

Comment:
I wonder if this logic would be more suitable to be defined directly in `_grpchacks` instead. For instance, defining a function `create_channel(channel_type, host, port, ssl)` where channel type is passed in and the `ChannelT` is returned

Reply:
I think it will make the code look more complicated, because then I need to define also `ChannelT` in `_grpchacks`. For now `_grpchacks` is only defining *types* for both libraries, it not trying to "merge" them, that's done separately (for example the `_exceptions` module also deals with exceptions from both libraries and unify them in the specific exceptions provided by the client. If I move this to `_grpchacks` it seems it would make sense to move the code to unify the exceptions too, and I think `_grpchacks` will get too big and with too much logic.

Reply:
Fair enough, I didn't consider the points you just mentioned. And I agree to keep `_grpchacks` minimal